src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -246,6 +289,18 @@ private void ProducingCompleteNoPreComplete(MemoryPoolIterator2 end)
             }
         }
 
+        private void CancellationTriggered()
+        {
+            lock (_contextLock)
+            {
+                // Abort the connection for any failed write
+                // Queued on threadpool so get it in as first op.
+                _connection?.Abort();
```

Comment:
Is `?.` for tests? If not we should just call `_connection.Abort` since `_connection` should always be set in the ctor.